backend/src/middleware/club.go

```diff
@@ -0,0 +1,55 @@
+package middleware
+
+import (
+	"github.com/GenerateNU/sac/backend/src/auth"
+	"github.com/GenerateNU/sac/backend/src/errors"
+	"github.com/GenerateNU/sac/backend/src/transactions"
+	"github.com/GenerateNU/sac/backend/src/types"
+	"github.com/GenerateNU/sac/backend/src/utilities"
+	"github.com/gofiber/fiber/v2"
+	"github.com/google/uuid"
+)
+
+func (m *MiddlewareService) ClubAuthorizeById(c *fiber.Ctx) error {
+	clubUUID, err := utilities.ValidateID(c.Params("id"))
+	if err != nil {
+		return errors.FailedToParseUUID.FiberError(c)
+	}
+
+	token, tokenErr := auth.ParseAccessToken(c.Cookies("access_token"))
+	if tokenErr != nil {
+		return errors.FailedToParseAccessToken.FiberError(c)
+	}
+
+	claims, ok := token.Claims.(*types.CustomClaims)
+	if !ok || !token.Valid {
+		return errors.FailedToValidateAccessToken.FiberError(c)
+	}
+
+	issuerUUID, issueErr := utilities.ValidateID(claims.Issuer)
+	if issueErr != nil {
+		return errors.FailedToParseAccessToken.FiberError(c)
+	}
+
+	// use club_id to get the list of admin for a certain club
+	clubAdmin, clubErr := transactions.GetAdminIDs(m.DB, *clubUUID)
+	if clubErr != nil {
+		return err
+	}
+
+	// check issuerID against the list of admin for the certain club
+	if ContainsUUID(clubAdmin, *issuerUUID) {
+		return c.Next()
+	}
+
+	return errors.Unauthorized.FiberError(c)
+}
+
+func ContainsUUID(uuids []uuid.UUID, targetUUID uuid.UUID) bool {
```

Comment:
can we replace with [slices.Contains](https://pkg.go.dev/slices#Contains)

Reply:
<img width="479" alt="Screenshot 2024-02-01 at 3 38 44 PM" src="https://github.com/GenerateNU/sac/assets/92384606/8cd43283-10fa-4106-9a94-61830bb62b23">